### ISSUE TITLE
Introduce new workflow field $participants

### ISSUE THREAD

**Issue by rsoika** [imixs/imixs-workflow#529]
Introduce a new internal workflow field named '$participants'.

* The field contains all users who had write access currently or in the past
* The field should be computed by the workflow engine after a process life-cycle.

**Comment by rsoika** [comment]
Hi @chrisw14 : what do you think about this new workflow item?

**Comment by chrisw14** [comment]
Hi Ralph,
to be honest, I already have exactly this field in my application :-)
It's a big advantage if you can use this field for read access. So all users which participates in the workflow, can check their approvals later again.
I think, I extended a plugin to implement this but maybe it's the better way to add this to the workflow engine.

**Comment by chrisw14** [comment]
This is my implementation:

```
	public ItemCollection updateParticipants(ItemCollection workitem) {
		
		List<String> participants = workitem.getItemValue("$participants");
		int step = workitem.getItemValueInteger("$processId");
		String user = loginController.getUserPrincipal();
		if (!participants.contains(user) && step > 1) {
			participants.add(user);
			workitem.replaceItemValue("$participants", participants);
		}
		
		return workitem;
	}
```

**Comment by rsoika** [comment]
good point: you only add the current user and not all users listed in the owner field (which was my first thought) . This makes sense, as some users listed in the owner field will never participate in a specific workflow. I think I will take your implementation.

**Comment by chrisw14** [comment]
Yes, for me it‘s enough to add only active participants. If there is a need for passive participants to have read access it‘s better to add them in the ACL of the Model I think.

**Comment by rsoika** [comment]
implemented by Issue #531
